SystemServices/SystemServices.cpp

```diff
@@ -500,6 +507,156 @@ namespace WPEFramework {
             returnResponse(result);
         }//end of requestSystemReboot
 
+        /*
+         * @brief This function delays the reboot in seconds.
+         * This will internally sets the tr181 fwDelayReboot parameter.
+         * @param1[in]: {"jsonrpc":"2.0","id":"3","method":"org.rdk.System.2.fwDelayReboot",
+         *                  "params":{"fwDelayReboot": int seconds}}''
+         * @param2[out]: {"jsonrpc":"2.0","id":3,"result":{"success":<bool>}}
+         * @return: Core::<StatusCode>
+         */
+
+        uint32_t SystemServices::fwDelayReboot(const JsonObject& parameters,
+                JsonObject& response)
+        {
+            bool result = false;
+            uint32_t delay_in_sec = 0;
+
+            if ( parameters.HasLabel("fwDelayReboot") ){
+                /* get the value */
+                delay_in_sec = static_cast<unsigned int>(parameters["fwDelayReboot"].Number());
+
+                /* we can delay with max 24 Hrs = 86400 sec */
+                if (delay_in_sec > 0 && delay_in_sec <= MAX_REBOOT_DELAY ){
+
+                    const char * set_rfc_val=(parameters["fwDelayReboot"].String()).c_str();
+
+                    LOGINFO("set_rfc_value %s\n",set_rfc_val);
+
+                    /*set tr181Set command from here*/
+                    WDMP_STATUS status = setRFCParameter((char*)"thunderapi",
+                            TR181_FW_DELAY_REBOOT, set_rfc_val, WDMP_INT);
+                    if ( WDMP_SUCCESS == status ){
+                        result=true;
+                        LOGINFO("Success Setting the fwDelayReboot value\n");
+                    }
+                    else {
+                        LOGINFO("Failed Setting the fwDelayReboot value %s\n",getRFCErrorString(status));
+                    }
+                }
+                else {
+                    /* we didnt get a valid Auto Reboot delay */
+                    LOGERR("Invalid FwDelayReboot Value Max.Value is 86400 sec\n");
```

Comment:
change to delaySeconds

Reply:
modified as _Invalid setFirmwareRebootDelay Value Max.Value is 86400 sec_